### PR TITLE
Improved Error in :mod:`.utils.tex_file_writing`

### LINKED ISSUE #2360: Tex value error when handling compilation issue
## Description of bug / unexpected behavior

I had some missing tex dependencies that are resolved now, but when manim was gathering the details to report the error it failed as `env_markers_indices` in `tex_file_writing` > `print_tex_error` method was empty. It instead through the value error below in the logs that creates some noise hiding the real issue. 

## Expected behavior

Expected behaviour is that it should be able to handle empty `env_markers_indices` if there are cases where it will be empty and report the reason for the latex compilation error and not throw a ValueError

## How to reproduce the issue

In my case I was missing latex dependencies that triggered this issue in ubuntu and ran the test

```
pytest manim/tests/test_graphical_units/test_axes.py::test_axes
```

<details><summary>Code for reproducing the problem</summary>

```py
Paste your code here.
```

</details>

## Additional media files

<details><summary>Images/GIFs</summary>

</details>

## Logs
<details><summary>Terminal output</summary>

```
[gw0] [100%] FAILED test_axes.py::test_axes (1, 480, 854, 4)
[12/04/21 08:58:14] INFO     Writing "x" to /tmp/pytest-o tex_file_writing.py:85
                             f-ryan/pytest-32/popen-gw0/t                       
                             est_axes0/049324c46070fc62.t                       
                             ex                                                 
                    ERROR    LaTeX compilation error:    tex_file_writing.py:265
                             LaTeX Error: File                                  
                             `standalone.cls' not found.                        
                                                                                

tests/test_graphical_units/test_axes.py:6 (test_axes)
scene = <manim.utils.testing._test_class_makers._make_test_scene_class.<locals>._TestedScene object at 0x7f4b94ce2f40>

    @frames_comparison
    def test_axes(scene):
        graph = Axes(
            x_range=[-10, 10, 1],
            y_range=[-10, 10, 1],
            x_length=6,
            y_length=6,
            color=WHITE,
        )
>       labels = graph.get_axis_labels()

test_axes.py:16: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../manim/mobject/coordinate_systems.py:437: in get_axis_labels
    self.get_x_axis_label(x_label),
../../manim/mobject/coordinate_systems.py:291: in get_x_axis_label
    return self._get_axis_label(
../../manim/mobject/coordinate_systems.py:391: in _get_axis_label
    label = self._create_label_tex(label)
../../manim/mobject/coordinate_systems.py:359: in _create_label_tex
    label_tex = MathTex(label_tex)
../../manim/mobject/svg/tex_mobject.py:304: in __init__
    raise compilation_error
../../manim/mobject/svg/tex_mobject.py:283: in __init__
    super().__init__(
../../manim/mobject/svg/tex_mobject.py:90: in __init__
    file_name = tex_to_svg_file(
../../manim/utils/tex_file_writing.py:46: in tex_to_svg_file
    dvi_file = compile_tex(
../../manim/utils/tex_file_writing.py:181: in compile_tex
    print_all_tex_errors(log_file, tex_compiler, tex_file)
../../manim/utils/tex_file_writing.py:253: in print_all_tex_errors
    print_tex_error(tex_compilation_log, error_index, tex)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

tex_compilation_log = ['This is pdfTeX, Version 3.14159265-2.6-1.40.21 (TeX Live 2020/Debian) (preloaded format=latex 2021.11.25)  4 DEC 202.../test_axes0/049324c46070fc62.tex\n', '(/tmp/pytest-of-ryan/pytest-32/popen-gw0/test_axes0/049324c46070fc62.tex\n', ...]
error_start_index = 9
tex_source = ['\\documentclass[preview]{standalone}\n', '\n', '\\usepackage[english]{babel}\n', '\\usepackage{amsmath}\n', '\\usepackage{amssymb}\n', '\n', ...]

    def print_tex_error(tex_compilation_log, error_start_index, tex_source):
        logger.error(
            f"LaTeX compilation error: {tex_compilation_log[error_start_index][2:]}",
        )
    
        # TeX errors eventually contain a line beginning 'l.xxx` where xxx is the line number that caused the compilation
        # failure. This code finds the next such line after the error current error message
        line_of_tex_error = (
            int(
                [
                    log_line
                    for log_line in tex_compilation_log[error_start_index:]
                    if log_line.startswith("l.")
                ][0]
                .split(" ")[0]
                .split(".")[1],
            )
            - 1
        )
        # our tex error may be on a line outside our user input because of post-processing
        if line_of_tex_error >= len(tex_source):
            return None
    
        # all lines numbers containing '\begin{' or '\end{' - except the Manim added center and document tags
        env_markers_indices = [
            idx
            for idx, log_line in enumerate(tex_source)
            if any(marker in log_line for marker in [r"\begin{", r"\end{"])
        ][2:-2]
    
        context = "Context of error:\n\n"
        if line_of_tex_error in env_markers_indices:
            context += "".join(tex_source[line_of_tex_error - 1 : line_of_tex_error + 1])
        else:
>           marker_before_error = max(
                idx for idx in env_markers_indices if idx < line_of_tex_error
            )
E           ValueError: max() arg is an empty sequence

../../manim/utils/tex_file_writing.py:298: ValueError








Coverage.py warning: Module manim was previously imported, but not measured (module-not-measured)


============================== 1 failed in 2.80s ===============================
Manim Community v0.11.0


Process finished with exit code 1
```

</details>

## System specifications

<details><summary>System Details</summary>

- OS (with version, e.g Windows 10 v2004 or macOS 10.15 (Catalina)): Ubuntu 21.10
- RAM: 16gb
- Python version (`python/py/python3 --version`): 3.9.7
- Installed modules (provide output from `pip list`):
```
PASTE HERE
```
</details>

<details><summary>LaTeX details</summary>

+ LaTeX distribution (e.g. TeX Live 2020): Version 3.14159265 (TeX Live 2020/Debian) (preloaded format=tex)
+ Installed LaTeX packages:

</details>

<details><summary>FFMPEG</summary>

Output of `ffmpeg -version`:

```
PASTE HERE
```
</details>

### PR DESCRIPTION
**Pull request by Kiran-Raj-Dev**
## Overview: What does this pull request change?
This removes the infamous `max() arg is an empty sequence` error
Added an insight for when a package is not installed
Fixes #2360

## Further Information and Comments

I am not sure if
> Install it using your LaTeX package manager

is a helpful enough insight. Suggestions Welcome!

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/utils/tex_file_writing.py [hunk 1]
```diff
@@ -149,6 +149,11 @@ def insight_inputenc_error(match):
     yield "See the documentation for manim.mobject.svg.tex_mobject for details on using a custom TexTemplate"
 
 
+def insight_package_not_found_error(match):
+    yield f"You do not have package {match[1]} installed"
```
Comment (Darylgolden) [inline manim/utils/tex_file_writing.py:153]:
```suggestion
    yield f"You do not have package {match[1]} installed."
```

File: manim/utils/tex_file_writing.py [hunk 2]
```diff
@@ -149,6 +149,11 @@ def insight_inputenc_error(match):
...
+    yield "Install it using your LaTeX package manager"
```
Comment (Darylgolden) [inline manim/utils/tex_file_writing.py:154]:
```suggestion
    yield "Install it using your LaTeX package manager."
```

**Review by Darylgolden — CHANGES_REQUESTED** [review]
Thank you! Can you add tests that assert the correct error messages are raised?

**Comment by Kiran-Raj-Dev** [comment]
Sure!

**Comment by Kiran-Raj-Dev** [comment]
I sank in enough time, but I have no idea why the logs will not be produced even after passing in the `--log_to_file` parameter.

**Comment by Kiran-Raj-Dev** [comment]
Insights are logged as INFO not ERROR.
Here the verbosity is set to ERROR. Since you asked:
> Can you add tests that assert the correct error messages are raised?

Setting it to INFO will include the insights

**Comment by Darylgolden** [comment]
I think it'll be good to test that the insight is raised.

File: manim/utils/tex_file_writing.py [hunk 3]
```diff
@@ -142,11 +142,16 @@ def tex_compilation_command(tex_compiler, output_format, tex_file, tex_dir):
...
     name = unicodedata.name(code_point)
-    yield f"TexTemplate does not support character '{name}' (U+{match[1]})"
-    yield "See the documentation for manim.mobject.svg.tex_mobject for details on using a custom TexTemplate"
+    yield f"TexTemplate does not support character '{name}' (U+{matching[1]})."
+    yield "See the documentation for manim.mobject.svg.tex_mobject for details on using a custom TexTemplate."
+
+
+def insight_package_not_found_error(matching):
+    yield f"You do not have package {matching[1]} installed."
+    yield f"Install {matching[1]} it using your LaTeX package manager, or check for typos"
```
Comment (Darylgolden) [inline manim/utils/tex_file_writing.py:154]:
```suggestion
    yield f"Install {matching[1]} it using your LaTeX package manager, or check for typos."
```

**Review by Darylgolden — CHANGES_REQUESTED** [review]

**Review by Darylgolden — APPROVED** [review]
Thank you, LGTM!

**Comment by behackl** [comment]
This unfortunately broke rendering of TeX strings containing `%`. Trying to create `Tex(r"\%")` fails in v0.15.0.

**Pull request merged**